goal_src/jak2/engine/target/board/target-board.gc

```diff
@@ -53,112 +53,66 @@
-    (while (< idx (-> this num-tricks-in-combo))
-      (let ((trick (-> this tricks-in-combo idx)))
-        (set! (-> freq trick) (inc (-> freq trick)))
-      (set! idx (inc idx))))
+    (dotimes (i (-> this num-tricks-in-combo))
+      (let ((trick (-> this tricks-in-combo i)))
+        (1+! (-> freq trick))))
+
     ;; Now we can construct the string
-    (set! idx 0)
-    (while (< idx (-> this num-tricks-in-combo))
-      (let* ((trick (-> this tricks-in-combo idx))
-             (times (-> freq trick))
-             (skip? (-> printed trick))
-             (color (case trick
-                      (((board-tricks board-boost) (board-tricks board-jump) (board-tricks board-duck-jump) (board-tricks board-quick-jump))
-                       "<COLOR_PINK>")
-                      (((board-tricks board-rail) (board-tricks board-rail-jump))
-                       "<COLOR_RED>")
-                      (((board-tricks board-spin) (board-tricks board-flip) (board-tricks board-board-flip) (board-tricks board-nosegrab))
-                       "<COLOR_PAD-TRIANGLE>")
-                      (((board-tricks board-board-spin) (board-tricks board-method) (board-tricks board-noseflip) (board-tricks board-kickflip))
-                       "<COLOR_MENU-FLAG-ON>")))
-             (padding (if (= idx 0) "" " ")))
-        (when (not skip?)
-          (if (> times 0)
-            (format *pc-encoded-temp-string* "~S~S~S<COLOR_DEFAULT> x ~D" padding color (board-trick->string trick) times)
-            (format *pc-encoded-temp-string* "~S~S~S<COLOR_DEFAULT>" padding color (board-trick->string trick)))
-          (set! (-> printed trick) #t)
-          (set! text-entries (inc text-entries))
-          ;; TODO - creates an extra space, fix later
-          (when (= (mod text-entries 4) 0)
-            (format *pc-encoded-temp-string* "~%"))
-          ))
-      (set! idx (inc idx))))
-  ;; Add points
-  (when (> (-> this points-in-combo) 0.0)
-    (format *pc-encoded-temp-string* "~%~%<COLOR_CREDITS-STAFF-TITLE-2>~D" (the int (-> this points-in-combo))))
+    (dotimes (i (-> this num-tricks-in-combo))
+      (let ((trick (-> this tricks-in-combo i)))
+        (unless (-> printed trick)
+          (let ((times (-> freq trick))
+                (color (case trick
+                          (((board-tricks board-boost) (board-tricks board-jump) (board-tricks board-duck-jump) (board-tricks board-quick-jump))
+                          "~9L")
+                          (((board-tricks board-rail) (board-tricks board-rail-jump))
+                          "~3L")
+                          (((board-tricks board-spin) (board-tricks board-flip) (board-tricks board-board-flip) (board-tricks board-nosegrab))
+                          "~26L")
+                          (((board-tricks board-board-spin) (board-tricks board-method) (board-tricks board-noseflip) (board-tricks board-kickflip))
+                          "~15L")))
```

Comment:
can you revert back to using the color codes instead as i did in my most recent PR? It makes the code much more readable